### ISSUE TITLE
Tests don't compile (needs new PR) / Casing problem

### ISSUE THREAD

**Issue by BringerOD** [binaryfog/NameParser#9]
This does not parse in .net core.  Not sure if you have this running for .net core yet.  Seems like you do, because it says it supports .net standard 1.6

```
            string fullName = "SR. John Henry William dela Vega, Jr Esq.";
            FullNameParser target = FullNameParser.Parse(fullName);
            string firstName = target.FirstName;
            string middleName = target.MiddleName;
            string lastName = target.LastName;
            string title = target.Title;
            string nickName = target.NickName;
            string suffix = target.Suffix;
            string displayName = target.DisplayName;
```

To test this, open a new .net core unit test project,  I am using MS Test,  then add your nuget package,  create a test,  and put this code in it.

The code runs,  but all the data is empty.

Also,  not sure if this is related,  but if you download the code and run all the tests in VS 2017,  most break,  only 4 test green.

Thanks,

Bryan

**Comment by TYoung86** [comment]
Sorry, yes, SR. is not a valid prefix... It would be a valid suffix. I know it's in the Readme.md, but I don't know of any real world examples. See at the bottom of the https://github.com/binaryfog/NameParser/blob/master/README.md, the strikethrough...

Not sure how many differences there are between my branch (https://github.com/TYoung86/NameParser) and this one, but all the tests run and succeed. I don't know how much review was done on the merge, as there were lots of changes, many conflicts.

The tests for the project are using xUnit.

**Comment by BringerOD** [comment]
TYoung86 does not compile

Severity	Code	Description	Project	File	Line	Suppression State
Error	CS0535	'TestFullNamePattern' does not implement interface member 'IFullNamePattern.Rx'	BinaryFog.NameParser.Tests	CXXXXX\NameParser-master\BinaryFog.NameParser.Tests\TestFullNamePattern.cs	2	Active

**Comment by TYoung86** [comment]
Yep, I had uncommitted code. Committed and pushed. Sorry about that.

No code in the main project changed though, here's my test output;
[![test-output](https://user-images.githubusercontent.com/6979049/27842870-d8177886-60db-11e7-8886-f911d8ac306e.png)](http://jsbin.com/leseturole)

**Comment by TYoung86** [comment]
I'll make another pull request later to fix the tests with the code I hadn't committed.

Hard to keep track of this project, day job.

Do you have a real-world scenario where SR. is used as a prefix, or was this just a gotcha for the test cases? I'll add it back in if you have one.

**Comment by BringerOD** [comment]
No real world issue with that name.  I was just using it as an example.  What prompted me was I entered a name with a suffix of "III" and it did not work,  so I download the code to check it out and the tests did not run,  then I looked at the read me and posted the issue.

So suffix "iii" works,  but not "III",  not sure if that matters?  

Thanks for looking at it.

**Comment by TYoung86** [comment]
Yeah that's not good, I'll look into it.

**Comment by binaryfog** [comment]
Issue fixed in version 2.1
